cloud/deployment/fromfile/fromfile.go

```diff
@@ -0,0 +1,380 @@
+	"github.com/astronomer/astro-cli/config"
+	"github.com/ghodss/yaml"
+)
+
+var (
+	errEmptyFile                      = errors.New("has no content")
+	errCreateFailed                   = errors.New("failed to create deployment with input")
+	errRequiredField                  = errors.New("missing required field")
+	errCannotUpdateExistingDeployment = errors.New("already exists")
+	errNotFound                       = errors.New("does not exist")
+)
+
+const (
+	jsonFormat = "json"
+)
+
+// Create takes a file and creates a deployment with the confiuration specified in the file.
+// inputFile can be in yaml or json format
+// It returns an error if any required information is missing or incorrectly specified.
+func Create(inputFile string, client astro.Client, out io.Writer) error {
+	var (
+		err                                           error
+		errHelp, clusterID, workspaceID, outputFormat string
+		dataBytes                                     []byte
+		formattedDeployment                           inspect.FormattedDeployment
+		createInput                                   astro.CreateDeploymentInput
+		existingDeployments                           []astro.Deployment
+		createdDeployment                             astro.Deployment
+		nodePools                                     []astro.NodePool
+		jsonOutput                                    bool
+	)
+
+	// get file contents as []byte
+	dataBytes, err = os.ReadFile(inputFile)
+	if err != nil {
+		return err
+	}
+	// return errEmptyFile if we have no dataBytes
+	if len(dataBytes) == 0 {
+		return fmt.Errorf("%s %w", inputFile, errEmptyFile)
+	}
+	// set outputFormat if json
+	jsonOutput = isJSON(dataBytes)
+	// unmarshal to a formattedDeployment
+	err = yaml.Unmarshal(dataBytes, &formattedDeployment)
+	if err != nil {
+		return err
+	}
+	// validate required fields
+	err = checkRequiredFields(&formattedDeployment)
+	if err != nil {
+		return err
+	}
+	c, err := config.GetCurrentContext()
+	if err != nil {
+		return err
+	}
+	// map workspace name to id
+	workspaceID, err = getWorkspaceIDFromName(formattedDeployment.Deployment.Configuration.WorkspaceName, c.Organization, client)
+	if err != nil {
+		return err
+	}
+	// map cluster name to id and collect node pools for cluster
+	clusterID, nodePools, err = getClusterInfoFromName(formattedDeployment.Deployment.Configuration.ClusterName, c.Organization, client)
+	if err != nil {
+		return err
+	}
+	existingDeployments, err = client.ListDeployments(c.Organization, workspaceID)
+	if err != nil {
+		return err
+	}
+	// check if deployment exists
+	if deploymentExists(existingDeployments, formattedDeployment.Deployment.Configuration.Name) {
+		// create does not allow updating existing deployments
+		errHelp = fmt.Sprintf("use deployment update --from-file %s instead", inputFile)
+		return fmt.Errorf("deployment: %s %w: %s", formattedDeployment.Deployment.Configuration.Name,
+			errCannotUpdateExistingDeployment, errHelp)
```

Comment:
nitpick: I don't see the flag being added as part of this PR for `astro deployment update`, is it planned to come in some upcoming PR?

Reply:
yes the scope is to create in this PR. Updates will be next.